core/base/src/TString.cxx

```diff
@@ -238,6 +244,12 @@ TString::~TString()
 
 char *TString::Init(Ssiz_t capacity, Ssiz_t nchar)
 {
+
+   if (capacity < 0 || nchar < 0)
+   {
+      char *data = 0; // sets pointer to zero 
+      return data;
```

Comment:
That effectively returns a `nullptr`, but we want to return an empty string.  I think the easiest would be to check `capacity` and `nchar` independently at the beginning and set them to zero if they are negative. The rest of the method should then take care of the correct logic. While at it, if `nchar > capacity`  it would make sense to cap `nchar`.

Reply:
Ahh ok

Reply:
>While at it, if `nchar > capacity` it would make sense to cap `nchar`.

So set nchar to 0 if it's greater than capacity, right?

Reply:
Set it to capacity

Reply:
> Set it to capacity

ok